DiscordBot/commands/osu/index.ts

```diff
@@ -1,22 +1,82 @@
-import { Message } from "discord.js";
-import { User } from "../../../Models/user";
+import { Message, MessageEmbed, MessageEmbedOptions } from "discord.js";
+import { OAuth, User } from "../../../Models/user";
 import { Command } from "../index";
+import { config } from "node-config-ts";
+import { User as APIUser } from "nodesu";
+import Axios from "axios";
 
-async function command (message: Message) {
-    const user = await User.findOne({
-        discord: {
-            userID: message.author.id,
-        },
-    });
-    if (!user) {
-        await message.channel.send("No user found in the corsace database for you! Please login to https://corsace.io with your discord and osu! accounts!");
-        return;
+const osuRegex = /osu\s+(.+)/i;
+
+async function command (m: Message) {
+    let user: User;
+    let apiUser: APIUser;
+    if (!osuRegex.test(m.content)) { // Querying themself
+        const userQ = await User.findOne({
+            discord: {
+                userID: m.author.id,
+            },
+        });
+        if (!userQ) {
+            await m.channel.send("No user found in the corsace database for you! Please login to https://corsace.io with your discord and osu! accounts!");
+            return;
+        }
+
+        const { data } = await Axios.get(`https://osu.ppy.sh/api/get_user?k=${config.osu.v1.apiKey}&u=${userQ.osu.userID}`);
+        apiUser = new APIUser(data[0]);
+        if (!userQ.country) {
+            userQ.country = apiUser.country.toString();
+            await userQ.save();
```

Comment:
i'd rather include country update in the migration, rather than query-and-save if doesn't exist. just imagine if we did that everytime we added a new field